pkg/gen/python-templates/yaml.py.mustache

```diff
@@ -109,10 +110,19 @@ def _parse_yaml_object(obj, opts: Optional[pulumi.ResourceOptions] = None,
     if not obj:
         return []
 
+    # Create a copy of opts to pass into potentially mutating transforms that will be applied to this resource.
+    if opts is not None:
+        opts = copy(opts)
+    else:
+        opts = {}
+
     # Allow users to change API objects before any validation.
     if transformations is not None:
         for t in transformations:
-            obj = t(obj)
```

Comment:
BTW - note that this code previous did `obj = t(obj)` which is not the same semantics as JavaScript, and is possibly quite surprising.  It's also not compatible with the change we want to make here.

I went ahead and changed it, though this is a breaking change.  Curious if there's feedback on whether to take the breaking change to align these, or to adjust the PR somehow else to avoid it?

Reply:
Yeah, I'd go with the breaking change. It was unintentional behavior introduced when I ported from the Node SDK, and isn't widely used yet. We had at least one user mention that it was surprising, so I'd been intending to fix it anyway.